03.api.md

```diff
@@ -358,92 +358,973 @@ jerry_api_gc ();
 ```
 
 
+# jerry_api_eval
+
+**Summary**
+Perform JavaScript `eval`.
+
+**Prototype**
+
+```c
+jerry_completion_code_t
+jerry_api_eval (const jerry_api_char_t *source_p,
+                size_t source_size,
+                bool is_direct,
+                bool is_strict,
+                jerry_api_value_t *retval_p);
+```
+
+- `source_p` - source code to evaluate;
+- `source_size` - length of the source code;
+- `is_direct` - whether to perform `eval` in "direct" mode (possible only if `eval` invocation is performed from native function, called from JavaScript);
+- `is_strict` - perform `eval` as it is called from "strict mode" code;
+- `retval_p` - value, returned by `eval` (output parameter);
+- returned value - completion code that indicates whether run performed successfully (`JERRY_COMPLETION_CODE_OK`), or an unhandled JavaScript exception occurred (`JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION`).
+
+**Example**
+
+```c
+{
+  jerry_api_value_t ret_val;
+
+  jerry_completion_code_t status = jerry_api_eval (str_to_eval,
+                                                   strlen (str_to_eval),
+                                                   false, false,
+                                                   &ret_val);
+}
+```
+
+**See also**
+
+- [jerry_api_value_t](#jerry_api_value_t)
+- [jerry_api_create_external_function](#jerry_api_create_external_function)
+- [jerry_external_handler_t](#jerry_external_handler_t)
+
+
 # jerry_api_value_t
 
 **Summary**
-The data type represents any JavaScript value that can be sent to or received from the engine.
+The data type represents any JavaScript value that can be sent to or received from the engine.
+
+Type of value is identified by `jerry_api_value_t::type`, and can be one of the following:
+
+- `JERRY_API_DATA_TYPE_VOID` - no return value
+- `JERRY_API_DATA_TYPE_UNDEFINED` - JavaScript undefined;
+- `JERRY_API_DATA_TYPE_NULL` - JavaScript null;
+- `JERRY_API_DATA_TYPE_BOOLEAN` - boolean;
+- `JERRY_API_DATA_TYPE_FLOAT32` - number;
+- `JERRY_API_DATA_TYPE_FLOAT64` - number;
+- `JERRY_API_DATA_TYPE_UINT32` - number;
+- `JERRY_API_DATA_TYPE_STRING` - string;
+- `JERRY_API_DATA_TYPE_OBJECT` - reference to JavaScript object.
+
+**Structure**
+
+```c
+typedef struct jerry_api_value_t
+{
+  jerry_api_data_type_t type;
+
+  union
+  {
+    bool v_bool;
+
+    float v_float32;
+    double v_float64;
+
+    uint32_t v_uint32;
+
+    jerry_api_string_t v_string;
+    jerry_api_object_t v_object;
+  } u;
+} jerry_api_value_t;
```

Comment:
IMHO, yes we do. `jerry_api_value_t` is the wrapper of `ecma_value_t`. See the API examples. The user have to write code that access the members of an api value, so we need to document what can be stored in it. But I think it's already there with some comments. Isn't this a duplicate?